index.js

```diff
@@ -112,6 +112,12 @@ const Twilio = {
         }
         _eventHandlers[type].get(handler).remove()
         _eventHandlers[type].delete(handler)
+    },
+    removeAllListeners(type) {
+        _eventHandlers[type].forEach(value => {
+            value.remove()
+        })
```

Comment:
the foreach introduces a race condition here where `removeAllListeners` might return before all event handlers are removed.

Converting it to a traditional for-loop or returning a resolved promise when done will resolve this issue.